### ISSUE TITLE
Adding panels is broken

### ISSUE THREAD

**Issue by davkal** [grafana/grafana#14667]
0. At least one datasource
1. Create dashboard and click "Edit panel"

Result: Blank page
Worse: Going back to dashboard shows an empty dashboard with no action to do (existing panels if you had any are not displayed either).

Errors in JS console:
```
Cannot read property 'meta' of undefined
at ProxyComponent.renderOptions (QueryOptions.tsx?22d9:101)
```

Looks like the whole React component gets unmounted because of this error. I suggest to add ErrorBoundaries to keep things alive.

**Comment by torkelo** [comment]
Yes, error handling is something Johannes started on before Christmas.

What datasources have you added? Which is default? Think this only happens when Loki is default datasource (or any datasource that does not support metrics is marked as default)

**Comment by davkal** [comment]
Correct. Loki as default.

**Comment by jschill** [comment]
Yea, the whole react tree gets unmounted when thrown errors are not being caught.

I have added a guard if datasource is undefined (such as in the Loki case). 

(ErrorBoundaries is a great idea, but a global one is not in the scope of what I'm doing. My component is reusable though, but we need ux and a way to "reset" an error.)
